flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -201,6 +203,28 @@ public void releasePartitions(Collection<ResultPartitionID> partitionIds) {
 		}
 	}
 
+	/**
+	 * Report unreleased partitions.
+	 *
+	 * <p>Partitions are released in the following cases:
```

Comment:
I don't believe this javadoc belongs on this method. It contains information about implementation details of various methods, and thus should likely be placed into the class javadoc.

Reply:
This is mostly a prep step for introducing a shuffle service interface, it is not intended only for netty implementation. We can think where it belongs when the interface is introduced. I would keep it here now because it is mostly relevant here atm to understand what is unreleased.

Reply:
Let's remove this doc comment for now in favour of more comprehensive description given in #8608.